### ISSUE TITLE
Delete deleted messages

### ISSUE THREAD

**Issue by fabianonline** [fabianonline/telegram_backup#63]
This is necessary to keep up with Telegram's TOS.

**Comment by FatKiwi** [comment]
Not good :/ Can you make it optional?

**Comment by fabianonline** [comment]
Nope. That would be a violation of Telegram's TOS which could lead to this tool being banned from communicating with Telegram...

**Comment by FatKiwi** [comment]
Why would they give this info in the API then? It should be their job to not allow it.

**Comment by Rapti** [comment]
This would make this entire tool pointless. Please at least make it easy to turn this off when compiling this program using my own API key so I can violate the TOS myself.

**Comment by DrakeTDL** [comment]
I agree

**Comment by FatKiwi** [comment]
It will be nice if we can do it with our own API key, I like this idea. Hope the dev considers it.

**Comment by fabianonline** [comment]
Well... My interpretation of the TOS is that it is also not okay to simply let the user disable this functionality in the settings.

But... I can do absolutely nothing against users taking the jar file, opening it like a zip file and modify... let's say... build.properties in there. ;-)

**Comment by SebiderSushi** [comment]
Are these TOS publicly available?

**Comment by bodqhrohro** [comment]
@SebiderSushi https://core.telegram.org/api/terms, section 1.4.

**Comment by SebiderSushi** [comment]
@bodqhrohro Thank You very much for that link!

In my interpretation it ist absolutely ok to keep TelegramBackup's bahavior in this context since Telegram Desktop by now offers a data export feature (the one implemented for EU citizens to comply with GDPR but which anyone can use) so having an "export" of a deleted message would in no way "interfere with the basic functionality of Telegram" since what Telegram Backup does is in some form a basic functionality now. In my eyes Telegram Backup should be seen as just another data export tool which is absolutely not a client and should be allowed to do anything that the official export tool does, too.

**Comment by bodqhrohro** [comment]
@SebiderSushi actually, this fear is initially contrived. There already exists, for example, telegram_purple, the plugin for Pidgin, which technically does not support deleting messages from history, because it's stored as HTML logs instead of a database. And there still was no any complaints against it. Telegram is a mobile-first messenger, and the terms, as it's obvious from mentioning stores, is targeting mobile apps. All that homemade opensource non-mobile stuff is almost out of scope, just like the Slippery Joe :)
